pkg/kubectl/cmd/apply.go

```diff
@@ -270,6 +262,25 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 				gracePeriod:   options.GracePeriod,
 			}
 
+			// Get the modified configuration of the object. Embed the result
+			// as an annotation in the modified configuration, so that it will appear
+			// in the patch sent to the server.
+			modified, err := kubectl.GetModifiedConfiguration(info, true, encoder)
```

Comment:
I'm afraid to touch this code (for more than I already did) as I do not fully understand what is going on with the encoders and how `VersionedObject` vs `Object` works :/